python/ray/experimental/sgd/utils.py

```diff
@@ -121,3 +126,16 @@ def update(self, val, n=1):
         self.sum += val * n
         self.count += n
         self.avg = self.sum / self.count
+
+
+def check_for_failure(remote_values):
+    unfinished = remote_values
+    success = False
+    try:
+        while len(unfinished) > 0:
+            finished, unfinished = ray.wait(unfinished)
+            finished = ray.get(finished)
+        success = True
```

Comment:
nit: would be cleaner to me to just `return True` here and `return False` at the end of the function but no strong preference